x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -413,6 +414,7 @@ export function endDataLoad(layerId, dataId, requestToken, data, meta) {
       meta,
       requestToken
     });
+    dispatch(setLayerDataLoadErrorStatus(layerId, null));
```

Comment:
The app only ever dispatch layer errors on dataLoadError. That's why we can safely replace any error on the layer when there is a data-request ends succesfully.

That would not be the case if layers could generate errors in multiple places, or have multiple errors in general.